File: springboot-backend/src/main/java/com/capstone/project/service/ContentService.java
package com.capstone.project.service;

import com.capstone.project.model.Content;

import java.util.List;

public interface ContentService {

    List<Content> getAllByCardId(int id);

    Content createContent(Content content);

    Content getContentById(int id);

    Content updateContent(int id, Content contentDetails);

    Boolean deleteContent(int id);
}

File: react-frontend/src/services/ContentService.js
import axios from 'axios'

const API_BASE_URL = process.env.REACT_APP_HOST

const createContent = (content) => {
    return axios.post(API_BASE_URL + '/contents', content, {
        headers: {
            Authorization: `Bearer ${localStorage.getItem('userToken')}`,
        },
    })
}

const updateContent = (id, contentDetails) => {
    return axios.put(API_BASE_URL + '/contents/' + id, contentDetails, {
        headers: {
            Authorization: `Bearer ${localStorage.getItem('userToken')}`,
        },
    })
}

const getAllByCardId = (cardId) => {
    return axios.get(API_BASE_URL + '/contentbycardid?id=' + cardId)
}

const ContentService = {
    createContent,
    updateContent,
    getAllByCardId,
}

export default ContentService
